python_ta/contracts/__init__.py

```diff
@@ -27,10 +27,19 @@ def new_setattr(self, name, value):
         super_class.__setattr__(self, name, value)
         curframe = inspect.currentframe()
         callframe = inspect.getouterframes(curframe, 2)
-        if callframe[1][3] not in wrapped.__dict__:
+        frame_members = dict((name, member) for name,
+                             member in inspect.getmembers(callframe[1].frame))
+        frame_locals = frame_members['f_locals']
+        if 'self' in frame_locals and self is frame_locals['self']:
```

Comment:
(assuming `.f_locals` yields a dictionary) You can use the `dict.get` method, which returns `None` when the key isn't in the dictionary.

But also since we're at the bottom of the function here, you can negate this condition and remove the `return` branch, doing something like

```python
if ...:
    check_invariants(self)
```

Reply:
Done